_posts/2019-05-04-go-style-error-handling-in-javascript.md

```diff
@@ -77,125 +76,132 @@ if you have different ways of handling the different exceptions you will have to
+In other to turn this try-catch expression into the golang error handling pattern I mentioned earlier, I implemented a custom fetch function.
 
 ```javascript
-async function fetchable(url){
+async function fetchable(url) {
   try {
     let res = await fetch(url);
     return [null, res];
   } catch (err) {
-    return [err, null]
+    return [err, null];
   }
 }
 ```
 
-To use this simple function you just easily call it like this.
+To use this function we can easily call it like this:
 
 ```javascript
-const [ex, res] = await fetchable("https://example.com");
-if(ex){
+const [ex, res] = await fetchable('https://example.com/api');
+if (ex) {
   // handle exception.
 }
-// do something with res as it will exit
+// do something with res and exit
 ```
 
-What essentially happens in this new function is, one of two objects are always returned and never both. If the `ex` is truthy, the `res` object will not exist, and vice versa. If the underlying function throws, it will be captured in `ex`. 
+What happens is, one of two objects are always returned and never both. If the `ex` is truthy, the `res` object will not exist, and vice versa. If the underlying function throws, it will be captured in `ex`.
 
-You can easily extend the `fetchable` function to take in more arguments. 
+You can easily extend the `fetchable` function to take in more arguments.
 
-You can also implement a utitlity functon that takes in a promise and turns it into this golang style error handling pattern like this. 
+You can also implement a utility function which takes a promise as an argument and returns a golang error handling pattern like this:
 
 ```javascript
-async function Executable(promise){
+async function Executable(promise) {
   try {
     let res = await promise();
     return [null, res];
   } catch (err) {
-    return [err, null]
+    return [err, null];
   }
 }
 ```
 
-You can easily use this function multiple times like this. 
+You can use this function multiple times like this:
 
 ```javascript
-const [ex1, res] = await Executable(fetch("https://example.com"));
-if(ex1){
+const [ex1, res] = await Executable(fetch('https://example.com/api'));
+if (ex1) {
   // handle exception.
 }
 // do something with res
 
-const [ex2, data] = await Executable(db.query("SELECT * FROM USERS"));
-if(ex2){
+const [ex2, data] = await Executable(db.query('SELECT * FROM USERS'));
+if (ex2) {
   // handle exception.
 }
 // do something with data
 ```
 
-Whith this approach you know exactly what you are getting where. The straigth fowardness also helps in easily understanding how error cases are handled in expressions that can err, withour relagating them to the bottom. 
+With this approach you know exactly what you are getting where. The straightfowardness also helps in easily understanding how error cases are handled in expressions that can err, without relagating them to the bottom.
```

Comment:
I'm not really sure what you are trying to say here but can you readjust the first sentence.